balancer/rls/internal/keys/builder.go

```diff
@@ -122,6 +124,10 @@ func (bm BuilderMap) RLSKey(md metadata.MD, path string) KeyMap {
 	return b.keys(md)
 }
 
+func BuilderMapEqual(a, b BuilderMap) bool {
```

Comment:
(Optional. Your call.)

Make it a method of `BuilderMap`? `cmp.Equal` will be able to use it: https://github.com/google/go-cmp/blob/v0.4.0/cmp/compare.go#L53

Reply:
I had tried that first. But the problem is that I'm using `cmp.Equal` inside this function, and that was causing an infinite recursion.